mathesar_ui/src/systems/data-explorer/urlSerializationUtils.ts

```diff
@@ -34,56 +47,68 @@ export function constructDataExplorerUrlToSummarizeFromGroup(
   return undefined;
 }
 
-export function constructQueryModelFromTerseSummarizationHash(
+export function constructQueryModelFromHash(
   hash: string,
 ): UnsavedQueryInstance {
+  const queryInstance: Mutable<UnsavedQueryInstance> = {};
   const terseSummarization = JSON.parse(
     Url64.decode(hash),
-  ) as TerseSummarization;
-  const groupedColumnId = terseSummarization.terseGrouping[0][0];
-  const groupedColumn = terseSummarization.columns.find(
-    (column) => column.id === groupedColumnId,
-  );
-  if (!groupedColumn) {
-    return {};
+  ) as Partial<TerseSummarization>;
+
+  if (terseSummarization.baseTableId) {
```

Comment:
I would like this function to be re-written.
1. We should not accept a TerseSummarization, we should have a new TerseQueryModel, which can contain summarization.
2. I would live to avoid using Mutable here (and preferably anywhere in the codebase). Use the types of properties directly to contruct each property and return a single UnsavedQueryInstance at the end.

I do think this can be done separately after the PR is merged in.

Reply:
Tracked in https://github.com/centerofci/mathesar/issues/1925